File: ejemplo.js
function rc4(key, plaintext) {
    var S = [];
    var j = 0;
    var out = "";

    // Key Scheduling Algorithm (KSA)
    for (var i = 0; i < 256; i++) {
        S[i] = i;
    }

    for (var i = 0; i < 256; i++) {
        j = (j + S[i] + key.charCodeAt(i % key.length)) % 256;
        [S[i], S[j]] = [S[j], S[i]];
    }

    // Pseudo-Random Generation Algorithm (PRGA)
    i = j = 0;
    for (var y = 0; y < plaintext.length; y++) {
        i = (i + 1) % 256;
        j = (j + S[i]) % 256;
        [S[i], S[j]] = [S[j], S[i]];
        var K = S[(S[i] + S[j]) % 256];
        out += String.fromCharCode(plaintext.charCodeAt(y) ^ K);
    }

    return out;
}

var key = "clave";
var plaintext = "texto plano";
var ciphertext = rc4(key, plaintext);
console.log("Cifrado: " + ciphertext);

File: ejemplo.cs
using System;

class RC4
{
    public static byte[] Encrypt(byte[] key, byte[] data)
    {
        byte[] S = new byte[256];
        byte[] K = new byte[256];
        byte[] result = new byte[data.Length];
        int i, j = 0, t;

        // Key Scheduling Algorithm (KSA)
        for (i = 0; i < 256; i++)
        {
            S[i] = (byte)i;
            K[i] = key[i % key.Length];
        }

        for (i = 0; i < 256; i++)
        {
            j = (j + S[i] + K[i]) & 255;
            byte temp = S[i];
            S[i] = S[j];
            S[j] = temp;
        }

        // Pseudo-Random Generation Algorithm (PRGA)
        i = j = 0;
        for (int x = 0; x < data.Length; x++)
        {
            i = (i + 1) & 255;
            j = (j + S[i]) & 255;
            byte temp = S[i];
            S[i] = S[j];
            S[j] = temp;
            t = (S[i] + S[j]) & 255;
            result[x] = (byte)(data[x] ^ S[t]);
        }

        return result;
    }

    static void Main()
    {
        byte[] key = System.Text.Encoding.ASCII.GetBytes("clave");
        byte[] plaintext = System.Text.Encoding.ASCII.GetBytes("texto plano");
        byte[] ciphertext = Encrypt(key, plaintext);
        Console.WriteLine("Cifrado: " + System.Text.Encoding.ASCII.GetString(ciphertext));
    }
}
